app/collections/settings.js

```diff
@@ -1,89 +1,91 @@
+import {displayToast} from '../notifications';
+
+const Settings = connectionToDB('settings');
+
+export async function saveGradeSystem(data) {
+    try {
+        await Settings.insert(data);
+    } catch (e) {
+        displayToast('updateFail');
+        console.log(e);
     }
+}
+
+export async function getAddressData() {
+    try {
+        const results = await Settings.find({});
 
-    return resolve(entry);
-}));
+        return results;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
 
-export const getSystemType = () => new Promise(resolve => Settings.find({}, (err, docs) => {
-    if (err) {
-        updateFailed();
+        return null;
     }
+}
 
-    return resolve(docs);
-}));
+export async function getSystemType() {
+    try {
+        const results = await Settings.find({});
+
+        return results;
+    } catch (e) {
+        displayToast('retrieveFail');
+        console.log(e);
+
+        return null;
+    }
+}
 
-const updateAddress = (previous, id) => {
+async function updateAddress(previous, id) {
     const {title, street, province, country, zip, city, year} = previous;
-    Settings.update(
-        {_id: id},
-        {$set: {title, street, province, country, zip, city, year}},
-        {},
-        err => {
-            if (err) {
-                saveFailed();
-            }
-            saveSuccessful();
-        }
-    );
-};
-
-const updateSystem = (previous, id) => {
+    try {
+        await Settings.update({_id: id}, {$set: {title, street, province, country, zip, city, year}}, {});
+    } catch (e) {
+        displayToast('updateFail');
+        console.log(e);
+    }
+}
+
+async function updateSystem(previous, id) {
     const {note, points, percent} = previous;
+    try {
+        await Settings.update({_id: id}, {$set: {note, points, percent}}, {});
+        displayToast('saveSuccess');
+    } catch (e) {
+        displayToast('saveFail');
+        console.log(e);
+    }
+}
 
-    Settings.update({_id: id}, {$set: {note, points, percent}}, {}, err => {
-        if (err) {
-            saveFailed();
-        }
-        saveSuccessful();
-    });
-};
+export async function updateGradeType(data) {
+    let results = await getSystemType();
+    try {
+        await updateSystem(data, results[0]._id);
+        results = await getSystemType();
 
-export const updateGradeType = async data => {
-    const setting = await getSystemType();
+        return results;
+    } catch (e) {
+        displayToast('updateFail');
+        console.log(e);
 
-    return new Promise(resolve => {
-        updateSystem(data, setting[0]._id);
+        return null;
+    }
+}
 
-        return resolve(getSystemType());
-    });
-};
+export async function addAddress(data) {
+    let results = await getAddressData();
 
-export const addAddress = async data => {
-    const setting = await getAddressData();
+    try {
+        await updateAddress(data, results[0]._id);
```

Comment:
@hcuffy what if  `results[0]` is undefined?

Ok, I will stop writing it, I suggest you go through the codebase and take a look into those cases :)

Reply:
fixed